lib/plugins/aws/package/compile/events/kafka.js

```diff
@@ -47,8 +47,28 @@ class AwsCompileKafkaEvents {
               minItems: 1,
               items: { $ref: '#/definitions/awsSecretsManagerArnString' },
             },
+            clientCertificateTlsAuth: {
+              type: 'array',
+              minItems: 1,
+              items: { $ref: '#/definitions/awsSecretsManagerArnString' },
+            },
+            serverRootCaCertificate: {
+              type: 'array',
+              minItems: 1,
+              items: { $ref: '#/definitions/awsSecretsManagerArnString' },
+            },
           },
           additionalProperties: false,
+          oneOf: [
+            { required: ['vpcSubnet', 'vpcSecurityGroup'] },
+            { required: ['saslPlainAuth'] },
+            { required: ['saslScram256Auth'] },
+            { required: ['saslScram512Auth'] },
+            {
+              required: ['clientCertificateTlsAuth'],
```

Comment:
@medikoo What do you think of such approach to validation of dependent properties? So far we've usually did this type of validation on logic-level instead of schema.

Reply:
In that form It'll create imprecise error message, stating simply that confiiration for `kafka` event is invalid, not providing a hint what's not right.

I propose to move this to inline validation, we do the same in other cases as that

Reply:
yes i did notice that about the error messages, they were a bit poor. happy to change the approach, do you have any examples I could follow?

Reply:
I agree with the suggestion @medikoo, thanks for your input here 👍 

As for the examples, something similar to https://github.com/serverless/serverless/blob/d52526bb6059ce20eba341c29ad5a2373c238624/lib/plugins/aws/provider.js#L2107 would be good 👍